src/middlewared/middlewared/plugins/reporting/rest.py

```diff
@@ -72,6 +72,9 @@ def calculated_metrics_count(self):
             len(self.middleware.call_sync('device.get_disks', False, True)),
             cpu_info()['core_count'],
             self.middleware.call_sync('interface.query', [], {'count': True}),
+            self.middleware.call_sync('pool.dataset.query', [], {'count': True, 'extra': {
```

Comment:
Instead of calling `pool.dataset.query`, we can call `query_imported_fast_impl` directly from `middlewared.utils.zfs
`